src/admin/shared/components/ContentPicker/ContentPicker.tsx

```diff
@@ -80,91 +67,111 @@ const ContentPicker: FunctionComponent<ContentPickerProps> = ({
+		}
+
+		if (!selectedItem) {
+			return null;
+		}
+
+		const value = get(selectedItem, 'value', null);
+
+		if (!get(value, 'value')) {
+			onSelect(null);
+			setCurrentItem(null);
+			console.error(
+				new CustomError('[Content Picker] - Selected item has no value.', null, {
+					selectedItem,
+				})
+			);
+			toastService.danger(
+				i18n.t(
+					'admin/shared/components/content-picker/content-picker___voor-deze-content-pagina-is-geen-pad-geconfigureerd'
+				),
+				false
+			);
+			return null;
+		}
+
+		onSelect(value);
+		setCurrentItem(selectedItem);
 	};
 
-	const renderGroupLabel = (data: any) => <span>{data.label}</span>;
+	const onChangeInput = (value: string) => {
+		setInput(value);
+		onSelect(parsePickerItem(get(currentType, 'value') as ContentPickerType, value));
+	};
 
-	const renderContentPickerControls = () => {
+	// render controls
+	const renderTypePicker = () => (
+		<ReactSelect
+			{...REACT_SELECT_DEFAULT_OPTIONS}
+			id="content-picker-type"
+			placeholder={t('Type')}
+			aria-label={i18n.t('Selecteer een type')}
+			options={typeOptions}
+			onChange={onSelectType}
+			value={currentType}
+			isSearchable={false}
+			isOptionDisabled={(option: PickerTypeOption) => !!option.disabled}
+			noOptionsMessage={() => i18n.t('Geen types')}
+		/>
+	);
+
+	const renderItemControl = () => {
 		if (!currentType) {
 			return null;
 		}
 
 		switch (currentType.picker) {
 			case 'SELECT':
-				return renderSelectPicker();
+				return renderItemPicker();
 			case 'TEXT_INPUT':
 				return renderTextInputPicker();
 			default:
 				return null;
 		}
 	};
 
-	const renderSelectPicker = () => (
-		<ReactSelect
+	const renderItemPicker = () => (
+		<AsyncSelect
 			{...REACT_SELECT_DEFAULT_OPTIONS}
-			id="content-picker-query"
-			placeholder={t('admin/content/components/content-picker/content-picker___item')}
-			formatGroupLabel={renderGroupLabel}
-			options={options as any}
-			isSearchable={false}
-			isLoading={loading}
-			onChange={(selectedItem: ValueType<PickerItem>) => {
-				const value = get(selectedItem, 'value', null);
-
-				if (!get(value, 'value')) {
-					onSelect(null);
-					setCurrentValue(null);
-					console.error(
-						new CustomError('Selected content in content picker does not have a value', null, {
-							selectedItem,
-						})
-					);
-					toastService.danger(
-						i18n.t(
-							'admin/shared/components/content-picker/content-picker___voor-deze-content-pagina-is-geen-pad-geconfigureerd'
-						),
-						false
-					);
-					return;
-				}
-
-				onSelect(value);
-				setCurrentValue(selectedItem);
-			}}
-			value={currentValue}
+			id="content-picker-item"
+			placeholder={t('Item')}
+			aria-label={i18n.t('Selecteer een item')}
+			loadOptions={inflatePicker}
+			onChange={onSelectItem}
+			value={currentItem}
+			defaultOptions={options as any}
+			isClearable
+			noOptionsMessage={() => i18n.t('Geen resultaten')}
+			loadingMessage={() => i18n.t('Laden...')}
```

Comment:
je kan hier de t functie van de component gebruiken, no need for the i18n.t functie
Idem voor andere vertalingen in deze component